wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -169,52 +144,101 @@ final class DeviceInfoViewModel: ObservableObject {
+        proteusID: String?,
         isSelfClient: Bool,
         userSession: UserSession,
         credentials: ZMEmailCredentials?,
-        getUserClientFingerprintUseCase: GetUserClientFingerprintUseCaseProtocol,
-        e2eIdentityProvider: E2eIdentityProviding,
-        mlsProvider: MLSProviding
+        gracePeriod: TimeInterval,
+        conversationId: Data?,
+        getE2eIdentityEnabled: GetIsE2EIdentityEnabledUsecaseProtocol,
+        getE2eIdentityCertificates: GetE2eIdentityCertificatesUsecaseProtocol
     ) -> DeviceInfoViewModel {
         return DeviceInfoViewModel(
-            title: userClient.model ?? "",
-            addedDate: userClient.activationDate?.formattedDate ?? "",
-            proteusID: userClient.proteusSessionID?.clientID.fingerprintStringWithSpaces.uppercased() ?? "",
+            title: title,
+            addedDate: addedDate,
+            proteusID: proteusID?.uppercased().fingerprintStringWithSpaces ?? "",
             isProteusVerificationEnabled: userClient.verified,
             actionsHandler: DeviceDetailsViewActionsHandler(
                 userClient: userClient,
                 userSession: userSession,
                 credentials: credentials,
-                e2eIdentityProvider: e2eIdentityProvider,
-                mlsProvider: mlsProvider,
-                saveFileManager: SaveFileManager(systemFileSavePresenter: SystemSavePresenter())
+                conversationId: conversationId,
+                saveFileManager: SaveFileManager(systemFileSavePresenter: SystemSavePresenter()),
+                mlsClientResolver: MLSClientResolver(),
+                getE2eIdentityEnabled: getE2eIdentityEnabled,
+                getE2eIdentityCertificates: getE2eIdentityCertificates
             ),
-            userSession: userSession,
-            getUserClientFingerprint: getUserClientFingerprintUseCase,
             userClient: userClient,
-            isSelfClient: isSelfClient
+            isSelfClient: isSelfClient,
+            gracePeriod: gracePeriod
         )
     }
 }
+
+extension E2eIdentityCertificate {
+
+    // current default days the certificate is retained on server
+    private var kServerRetainedDays: Double { 28 * 24 * 60 * 60 }
+
+    // Randomising time so that not all clients update certificate at the same time
+    private var kRandomInterval: Double { Double(Int.random(in: 0..<86400)) }
+
+    var isExpired: Bool {
+        return expiryDate > Date.now
+    }
+
+    var isValid: Bool {
+        status == .valid
+    }
+
+    var isActivated: Bool {
+        return notValidBefore <= Date.now
+    }
+
+    var lastUpdatedDate: Date {
+        return notValidBefore + kServerRetainedDays + kRandomInterval
+    }
+
+    func shouldCertificateBeUpdated(with gracePeriod: TimeInterval) -> Bool {
+        guard isActivated,
+              isExpired,
+              (lastUpdatedDate + gracePeriod) < Date.now else {
+            return false
+        }
+        return true
+    }
```

Comment:
`suggestion:` 
1. we can simplify this by:
```
func shouldCertificateBeUpdated(with gracePeriod: TimeInterval) -> Bool {
        return isActivated,
                   !isExpired,
                   (lastUpdatedDate + gracePeriod) < Date.now 
    }
```

2. I would rename it to `shouldUpdate(with gracePeriod: TimeInterval)`. This is a method in the `E2eIdentityCertificate`, so there is no need to add "Certificate" to the method name.